### PR TITLE
Feature/process upload response

### PR DESCRIPTION
**Pull request by ioangut**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

* Handle different EFGS responses for Upload Job as per [jira item](https://jira.itc.sap.com/browse/EXPOSUREBACK-160).

### CODE REVIEW

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java [hunk 1]
```diff
@@ -69,6 +71,18 @@ public Upload(
...
 
+  private List<DiagnosisKey> executeFederationUpload(UploadPayload payload) {
+    logger.info("Executing batch request(s): {}", payload.getBatchTag());
+    var result = this.federationUploadClient.postBatchUpload(payload);
+    var retryKeys = result.getStatus500()
+        .stream()
+        .map(Integer::parseInt)
+        .map(index -> payload.getOrderedKeys().get(index))
+        .collect(Collectors.toList());
+    logger.info("Error on {} keys, marking them for retry", retryKeys.size());
```
Comment (sorin-stefan-iovita-sap) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java:82]:
```suggestion
    logger.error("Error on {} keys, marking them for retry", retryKeys.size());
```
or use logger.warn

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/payload/UploadPayload.java [hunk 2]
```diff
@@ -42,6 +41,7 @@
   private List<FederationUploadKey> originalKeys;
   private String batchSignature;
   private String batchTag;
+  private List<DiagnosisKey> orderedKeys;
```
Comment (EugenM-SAP) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/payload/UploadPayload.java:44]: Not sure if it's needed to store a copy of the keys in the upload payload. Rather whenever we need the ordered list we can on demand sort the one in the batch. This however is not crucial and can be dealt with in a later refactoring step.

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java [hunk 3]
```diff
@@ -69,6 +72,18 @@ public Upload(
     this.uploadKeyService = uploadKeyService;
   }
 
+  private List<DiagnosisKey> executeFederationUpload(UploadPayload payload) {
```
Comment (EugenM-SAP) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java:75]: suggest to use executeUploadAndCollectErrors(). as a method name to make it clear that the returned list is a collection of rejected keys.
Or maybe split the extraction of rejected keys in a separate method and just return the BatchUploadResponse structure from this method

Comment (EugenM-SAP) [reply services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java:75]: I think if we keep the original keys sorted by the same criteria (keyData) we can just collect the indexes from the upload response, and pass those to the markSuccesful...() method as a parameter. Something like ignoreIndexes.

**Comment by sorin-stefan-iovita-sap** [comment]
> SonarCloud Quality Gate failed.
> 
> [Bug](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=VULNERABILITY) (and [Security Hotspot](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=SECURITY_HOTSPOT) [1 Security Hotspot](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=SECURITY_HOTSPOT) to review)
> [Code Smell](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=CODE_SMELL) [3 Code Smells](https://sonarcloud.io/project/issues?id=corona-warn-app_cwa-server&pullRequest=784&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=corona-warn-app_cwa-server&pullRequest=784&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=corona-warn-app_cwa-server&pullRequest=784&metric=new_coverage&view=list)
> [](https://sonarcloud.io/component_measures?id=corona-warn-app_cwa-server&pullRequest=784&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=corona-warn-app_cwa-server&pullRequest=784&metric=new_duplicated_lines_density&view=list)

Does anyone know why is the coverage 0%? This seems strange since there are tests that covers a part of this code.

**Review by EugenM-SAP — APPROVED** [review]

**Review by sorin-stefan-iovita-sap — APPROVED** [review]

**Pull request merged**